benchmarks/dynamo/common.py

```diff
@@ -375,6 +377,116 @@ def output_csv(filename, headers, row):
             writer.writerow(list(line) + ["0"] * (len(headers) - len(line)))
 
 
+def get_suite_from_model_iter_fn(model_iter_fn):
+    # TODO: This is a bit of a hack
+    suite = None
+    if (runner := getattr(model_iter_fn, "__self__", None)) and hasattr(
+        runner, "suite_name"
+    ):
+        suite = runner.suite_name
+    return suite
+
+
+def output_signpost(data, args, suite, error=None):
+    from torch.utils._stats import simple_call_counter
+
+    data = data.copy()
+
+    if "name" not in data:
+        data["name"] = current_name
+
+    if "dev" not in data:
+        data["dev"] = current_device
+
+    filtered_args = vars(args).copy()
+    # I generated this list by reading through all the configs and dropping
+    # ones that looked irrelevant or redundant
+    for k in [
+        "filter",
+        "exclude",
+        "exclude_exact",
+        "dump_raw_metrics",
+        "log_operator_inputs",
+        "distributed_master_port",
+        "skip_accuracy_check",
+        "generate_aot_autograd_stats",
+        "output",
+        "output_directory",
+        "disable_output",
+        "export_profiler_trace",
+        "profiler_trace_name",
+        "explain",
+        "stats",
+        "print_memory",
+        "print_compilation_time",
+        "print_dataframe_summary",
+        "print_graph_breaks",
+        "log_graph_breaks",
+        "timing",
+        "progress",
+        "timeout",
+        "per_process_memory_fraction",
+        "minify",
+        "verbose",
+        "quiet",
+        "print_fx",
+        "print_aten_ops",
+        "log_conv_args",
+        "recompile_profiler",
+        "find_batch_sizes",
+        # Redundant
+        "batch_size",
+        "batch_size_file",
+        "only",
+        "diff_branch",
+        "tag",
+        "coverage",
+        "overhead",
+        "speedup_dynamo_ts",
+        "speedup_fx2trt",
+        "speedup_fx2trt_fp16",
+        "accuracy",
+        "performance",
+        "tolerance",
+    ]:
+        del filtered_args[k]
+
+    event_name = "unknown"
+    if args.accuracy:
+        event_name = "accuracy"
+    elif args.quantization:
+        event_name = "quantization"
+    elif args.performance:
+        event_name = "performance"
+
+    from torch._dynamo.utils import calculate_time_spent, compilation_time_metrics
+
+    open_source_signpost(
```

Comment:
in hack land, we consolidated all logging in one place. it made it really nice b/c if you were doing analysis you could see all the different event types and how they are constructed. don't need to change anything here but just food for thought: https://www.internalfb.com/code/fbsource/[75e75d47d29c8bebd337dc1630341bfe43afb2b3]/fbcode/hphp/hack/src/logging/facebook/hackEventLogger.ml

Reply:
Interesting. What would you say was important to be in hackEventLogger.ml? Because you still have to actually call into "logging" from all the relevant call sites. Is it just the actual JSON construction?

Reply:
I think one of the big benefits was greppability. For example if I wanted to know how saved state telemetry was logged, I just ctrl F for saved_state in that file. Or if I want to add a similar column as another event, I could grep for callsites of that particular event and easily add that column to my event.

@jamesjwu is also quite familiar with this paradigm and can probably chime in w/ his own experience.

Reply:
Seems like it might be possible to do for us. One troublesome piece is we don't uniformly have type checking everywhere, so functions with large amounts of positional arguments do not necessarily work super well.